estimagic/dashboard/monitoring_callbacks.py

```diff
@@ -162,36 +165,48 @@ def _update_monitoring_tab(
     missing = [i for i, val in enumerate(data["value"]) if val is None]
     crit_data = {
         "iteration": [id_ for i, id_ in enumerate(data["rowid"]) if i not in missing],
-        "criterion": [val for i, val in enumerate(data["value"]) if i not in missing],
+        "criterion": [
+            np.clip(val, -clip_bound, clip_bound)
+            for i, val in enumerate(data["value"])
+            if i not in missing
+        ],
     }
     criterion_cds.stream(crit_data, rollover=rollover)
 
     # update the parameter plots
-    param_names = start_params["name"].tolist()
-    params_data = _create_params_data_for_update(data, param_names)
-    param_cds.stream(params_data, rollover=rollover)
+    # Note: we need **all** parameter ids to correctly map them to the parameter entries
+    # in the database. Only after can we restrict them to the entries we need.
+    param_ids = start_params["id"].tolist()
+    params_data = _create_params_data_for_update(data, param_ids, clip_bound)
+    available_keys = param_cds.data.keys()
```

Comment:
As discussed, I am in favor of defining a `stream_data` function that does the reduction automatically instead of calling `cds.stream` with a reduced dictionary, i.e:
```python
def stream_data(cds, data):
    # reduce data to present columns
    # call cds.stream with reduced data
```

Reply:
Done in 0afeae5.